pkg/bmc/ilo4.go

```diff
@@ -0,0 +1,88 @@
+package bmc
+
+import (
+	"net/url"
+)
+
+func init() {
+	registerFactory("ilo4", newILOAccessDetails, []string{})
```

Comment:
I don't know about the iLO protocol. Does it use HTTP? Do we need separate options for HTTP and HTTPS like we have with other protocols?

Reply:
i have made the changes.